client/src/shared/planCard/index.jsx

```diff
@@ -7,53 +7,340 @@ import { Grid } from '@material-ui/core';
+
+								return <Tag key={i} ref={tagsRef.current[i]}>
+									{tagIconUrl && <TagIcon src={tagIconUrl} />}
+									<Text
+										size='16px'
+										weight='normal'
+										text={tag.text}
+										color={theme.palette.black}
+									/>
+								</Tag>;
+							})}
+						</Tags>
+					</SC.CardContent>
 				</Link>
 			</SC.Card>
 		</Grid>
 	);
 };
-	
-
 
 
 PlanCard.propTypes = {
 	plan: PropTypes.object.isRequired,
 };
 
-export default PlanCard;
\ No newline at end of file
+export default PlanCard;
+
+const MapTitle = styled.div`
+    position: absolute;
+    width: 100%;
+    display: flex;
+    justify-content: space-between;
+    flex-direction: row;
+    align-items: center;
+    z-index: 9999;
+    padding: 13px 15px;
+`;
+
+const Chip = styled.div`
+    display: flex;
+    align-items: center;
+    background: #FFFFFF;
+    box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.2);
+    border-radius: 4px;
+    padding: 4px 9px;
+`;
+
+const StatusChip = styled(Chip)`
+    padding: 4px 7px;
+`;
+
+const StatusDot = styled.div`
+    width: 14px;
+    height: 14px;
+    flex-shrink: 0;
+    margin-left: 7px;
+    border-radius: 7px;
+    background: ${({ approved }) => approved ? '#1976D2' : '#AE7FF0'};
+`;
+
+const BookmarkBtn = styled.button`
+    width: 37px;
+    height: 37px;
+    border: none;
+    border-radius: 18.5px;
+    box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.2);
+    background: center no-repeat url(${({isBookmarked}) => isBookmarked ? BookmarkFilledIcon : BookmarkOutlinedIcon}) #FFFFFF;
+    flex-shrink: 0;
+    
+    & :focus {
+        outline: none;
+        background-color: #F5F5F5;
+    }
+`;
+
+const MapFooter = styled(MapTitle)`
+    bottom: 0;
+    z-index: 99999;
+`;
+
+const FooterChip = styled(Chip)`
+    padding: 4px 9px;
+`;
+
+const PlanDetailsHeader = styled.div`
+    display: flex;
+    flex-flow: row;
+    align-items: center;
+    height: 36px;
+    flex-shrink: 0;
+`;
+
+const PlanDistance = styled.div`
+    display: flex;
+    align-items: center;
+
+    &:before {
+        margin-left: 14px;
+        content: url(${DropPinIcon});
+    }
+
+    ${({ showDivider }) => showDivider && `
+      &:after {
+        width: 4px;
+        height: 4px;
+        border-radius: 50%;
+        background-color: black;
+        content: ' ';
+        margin: 10px;
+      }
+    `}
+`;
+
+const PlanName = styled.div`
+    flex-shrink: 0;
+    height: 50px;
+    overflow: hidden;
+  margin-bottom: 27px;
+`;
+
+const Tags = styled.div`
+    display: flex;
+    align-items: center;
+    flex-flow: wrap;
+    overflow: hidden;
+`;
+
+const Tag = styled.div`
+    display: flex;
+    align-items: center;
+    height: 26.35px;
+    background: #F5F5F5;
+    border-radius: 4px;
+    padding: 0 8px;
+    margin: 0 0 10px 10px;
+    
+    ${({ type }) => type && `
+      &:before {
+        width: 20px;
+        height: 20px;
+        // content: url(${require('../../assets/'+type+'-tag-icon.svg')});
```

Comment:
Please use the index file for the requires

Reply:
done 14529a8a61e5337e4846adf586386e4e1bc3ae0c